spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -272,10 +273,25 @@ public long count(Class<?> entityClass) {
 			resultsIterable = new DatastoreResultsIterable<>(convertEntitiesForRead(results, entityClass),
 					results.getCursorAfter());
 		}
+		if (query instanceof StructuredQuery) {
+			resultsIterable.setHasNextPageQuery(
```

Comment:
Instead of calling `setHasNextPageQuery` can this block of code just move to `DatastoreResultsIterable`?
Also, why does this only apply to `StructuredQuery`?